### ISSUE TITLE
Reproducing on single seed value

### ISSUE THREAD

**Issue by surbhi1944** [hill-a/stable-baselines#557]
Even after setting seed everywhere. When i run the same code (training DDPG) 20 times. Evaluation results on all 20 models are different. hence results on single seed are not  stable. how to fix them?  The attached file represents evaluation results on 10^5 steps.

The way i seed the libraries are as follow:
    import gym
    import random
    import numpy as np
    import os
    import matplotlib.pyplot as plt
    import tensorflow as tf
    from stable_baselines.ddpg.policies import MlpPolicy
    from stable_baselines.bench import Monitor
    from stable_baselines.results_plotter import load_results, ts2xy
    from stable_baselines import DDPG
    from stable_baselines.common.noise import OrnsteinUhlenbeckActionNoise
    from datetime import datetime, timedelta
    import time
    from mpi4py import MPI

    #from stable_baselines.common.misc_util import set_global_seeds
    from stable_baselines import logger
    from stable_baselines.common import set_global_seeds
    # Configure things.
    rank = MPI.COMM_WORLD.Get_rank()
    if rank != 0:
        logger.set_level(logger.DISABLED)
    seed=12
    set_global_seeds(seed)
    seed = seed + 1000000 * rank
    os.environ['PYTHONHASHSEED']=str(seed)
    logger.info('rank {}: seed={}, logdir={}'.format(rank, seed, logger.get_dir()))
    random.seed(seed) 
    np.random.seed(seed)

    tf.set_random_seed(seed)
    from keras import backend as K
    session_conf=tf.ConfigProto(intra_op_parallelism_threads=1,inter_op_parallelism_threads=1)
    sess=tf.Session(graph=tf.get_default_graph(),config=session_conf)
    K.set_session(sess)
    tf.reset_default_graph()
    env = gym.make('Walker2d-v3')
    env.seed(seed)
 
[normal_eval (copy).txt](https://github.com/hill-a/stable-baselines/files/3851137/normal_eval.copy.txt)

  How to fix results on single seed value?

**Comment by araffin** [comment]
Hello,

Please fill the issue template completely and read the documentation.
Related issue: https://github.com/hill-a/stable-baselines/issues/539

I may close this one as it seems to be a duplicate anyway.

**Comment by surbhi1944** [comment]
Thanks for reply. I have done seeding at every place: 

model = DDPG(MlpPolicy, env, param_noise=None,batch_size=64,buffer_size=1000000,enable_popart=False, action_noise=action_noise, verbose=4)

model.learn(total_timesteps=int(1e6),seed=seed,callback=callback,log_dir=log_dir)

Still not getting fixed results on same seed. How can i use these results for research purpose?

**Comment by araffin** [comment]
If you would have filled the issue template, it would have been obvious you were not using the master version (so you are using v<=2.8.0).
As mentioned in #539, you should either wait for 2.9 release or use the master version to have reproducible results.
Again, please read the [documentation](https://stable-baselines.readthedocs.io/en/master/guide/algos.html#reproducibility).

Closing this issue for the above reasons.
